File: ios/Classes/coins/SOL.swift
//
//  SOL.swift
//  trustdart
//
//  Created by Jay on 3/29/22.
//
import WalletCore

class SOL: Coin  {
    init() {
        super.init(name: "SOL", coinType: .solana)
    }
    
    override func getPublicKey(path: String, mnemonic: String, passphrase: String) -> String? {
        let wallet = HDWallet(mnemonic: mnemonic, passphrase: passphrase)
        return wallet!.getKey(coin: self.coinType, derivationPath: path).getPublicKeyEd25519().data.base64EncodedString()
    }

    override func getRawPublicKey(path: String, mnemonic: String, passphrase: String) -> FlutterStandardTypedData {
        let wallet = HDWallet(mnemonic: mnemonic, passphrase: passphrase)
        return FlutterStandardTypedData(bytes: wallet!.getKey(coin: self.coinType, derivationPath: path).getPublicKeyEd25519().data)
    }
    
}

File: android/src/main/kotlin/africa/ejara/trustdart/coins/SOL.kt
import africa.ejara.trustdart.Coin
import africa.ejara.trustdart.utils.base64String
import africa.ejara.trustdart.utils.toHex
import android.util.Base64
import wallet.core.jni.CoinType
import wallet.core.jni.Curve
import wallet.core.jni.HDWallet


class SOL : Coin("SOL", CoinType.SOLANA) {
    override fun getPublicKey(path: String, mnemonic: String, passphrase: String): String? {
        val wallet = HDWallet(mnemonic, passphrase)
        return wallet.getKey(coinType, path).publicKeyEd25519.data().base64String()
    }

    override fun getRawPublicKey(path: String, mnemonic: String, passphrase: String): ByteArray? {
        val wallet = HDWallet(mnemonic, passphrase)
        return wallet.getKey(coinType, path).publicKeyEd25519.data()
    }
}